doc/howtos/rdtraining/basicviews.rst

```diff
@@ -0,0 +1,220 @@
+      :align: center
+      :alt: Form view
+
+Forms are used to create and edit single records.
+
+Their root element is ``<form>``. They are composed of high-level structure
+elements (groups, notebooks) and interactive elements (buttons and fields):
+
+.. code-block:: xml
+
+    <form string="Test">
+        <sheet>
+            <group>
+                <group>
+                    <field name="name"/>
+                </group>
+                <group>
+                    <field name="last_seen"/>
+                </group>
+                <notebook>
+                    <page string="Description">
+                        <field name="description"/>
+                    </page>
+                </notebook>
+            </group>
+        </sheet>
+    </form>
+
+It is possible to use regular HTML tags such as ``div`` and ``h1``, and some ``class`` are
+available to fine-tune the look.
+
+A simple example can be found
+`here <https://github.com/odoo/odoo/blob/6da14a3aadeb3efc40f145f6c11fc33314b2f15e/addons/crm/views/crm_lost_reason_views.xml#L16-L44>`__.
+
+.. exercise:: Add a custom form view
+
+    Define a form view for the ``estate.property`` model in the appropriate XML file. Check the
+    **Goal** of the section for the final design of the page.
+
+This might require some trial and errors before you get to the expected result ;-) It is advised
+to add the fields and the tags one at a time to understand how it works.
+
+Search
+======
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/views/search`.
+
+.. note::
+
+    **Goal**: at the end of this section, the search view should look like this:
+
+    .. image:: basicviews/media/search_01.png
+      :align: center
+      :alt: Search fields
+
+    .. image:: basicviews/media/search_02.png
+      :align: center
+      :alt: Filter
+
+    .. image:: basicviews/media/search_03.png
+      :align: center
+      :alt: Group By
+
+Search views are slightly different from the list and form views since they don't display
+*content*: although they apply to a specific model, they are used to filter
+other view's content (generally aggregated views such as :ref:`reference/views/list`).
+Beyond that difference in use case, they are defined the same way.
+
+Their root element is ``<search>``. The simplest form of the tree view simply
+lists all the fields a shortcut is needed:
+
+.. code-block:: xml
+
+    <search string="Tests">
+        <field name="name"/>
+        <field name="last_seen"/>
+    </search>
+
+The default search view generated by Odoo provides a shortcut to filter by ``name``. It is very
+common to add the fields on which the user is likely to filter in a customized search view.
+
+Search view ``<field>`` elements can have a ``@filter_domain`` that overrides
+the domain generated for searching on the given field. In the given domain,
+``self`` represents the value entered by the user. In the example below, it is
```

Comment:
which example?

Reply:
Not sure if this paragraph is needed, the filter is explained better in the paragraph after the exercise and the introduction of domain is better in the next paragraph.

Reply:
wrong copy-paste, paragraph removed